akka-http/src/main/scala/akka/http/javadsl/server/Rejections.scala

```diff
@@ -101,6 +101,14 @@ trait MalformedHeaderRejection extends Rejection {
 }
 
 /**
+ * Rejection created by [[akka.http.scaladsl.server.directives.HeaderDirectives.checkSameOrigin]].
+ * Signals that the request was rejected because `Origin` header value is invalid.
+ */
+trait InvalidOriginHeaderRejection extends Rejection {
+  def headerValue: String
```

Comment:
well, then `getOrigin` here, for the JavaDSL one, and in the scaladsl one `val origin: Origin` and implement the getOrigin as `override def getOrigin = origin`